packages/common/src/common.ts

```diff
@@ -307,13 +307,19 @@ export class Common extends EventEmitter {
    * @param td : total difficulty of the parent block (for block hf) OR of the the chain latest (for chain hf)
    * @returns The name of the HF
    */
-  getHardforkByBlockNumber(blockNumber: BigIntLike, td?: BigIntLike): string {
+  getHardforkByBlockNumber(
+    blockNumber: BigIntLike,
+    td?: BigIntLike,
+    timestamp?: BigIntLike
+  ): string {
     blockNumber = toType(blockNumber, TypeOutput.BigInt)
     td = toType(td, TypeOutput.BigInt)
+    timestamp = toType(timestamp, TypeOutput.Number)
 
-    // Filter out hardforks with no block number and no ttd (i.e. unapplied hardforks)
+    // Filter out hardforks with no block number, no ttd or no timestamp (i.e. unapplied hardforks)
     const hfs = this.hardforks().filter(
-      (hf) => hf.block !== null || (hf.ttd !== null && hf.ttd !== undefined)
+      (hf) =>
+        hf.block !== null || (hf.ttd !== null && hf.ttd !== undefined) || hf.timestamp !== undefined
```

Comment:
unless someone specifies them as null and tries to use them outside typescript since typescript would complain while trying to match with `HardforkConfig` type.
But open to having/allowing them as null and introduce appropriate checks, but may be as a separate PR